core/controllers/topic_editor.py

```diff
@@ -149,8 +151,44 @@ def get(self, topic_id: str) -> None:
         additional_story_summary_dicts = [
             summary.to_dict() for summary in additional_story_summaries]
 
+        canonical_stories_ids = [summary['id'] for
+            summary in canonical_story_summary_dicts]
+        canonical_stories = list(filter(
+            None, story_fetchers.get_stories_by_ids(canonical_stories_ids)))
+        canonical_stories_dict: Dict[str, story_domain.Story] = {
+            canonical_story.id: canonical_story for canonical_story in
+            canonical_stories}
         updated_canonical_story_summary_dicts = []
+
         for summary in canonical_story_summary_dicts:
+            if summary['id'] not in canonical_stories_dict:
+                continue
+            story = canonical_stories_dict[summary['id']]
+            nodes = story.story_contents.nodes
+            total_chapters_count = len(nodes)
+            published_chapters_count = 0
+            upcoming_chapters_count = 0
+            overdue_chapters_count = 0
+            upcoming_chapters_expected_days = []
+            for node in nodes:
+                if node.status == constants.STORY_NODE_STATUS_PUBLISHED:
+                    published_chapters_count += 1
+                if (node.status != constants.STORY_NODE_STATUS_PUBLISHED and
+                    node.planned_publication_date is not None):
+                    current_time = (
+                        utils.convert_millisecs_time_to_datetime_object(
```

Comment:
@Kej-r03 sorry, one more question. Why not use `current_time = datetime.datetime.now()`? It also returns a timezone unaware object in your machine's local timezone

Reply:
Yes I had done that in my last PR, due to which it got reverted, because the test was becoming dependent on datetime.now() which changes with time, so initially it passed, but later on it started failing.

Reply:
How is that different than the current code? We are still using `datetime.datetime.utcnow()` to get the current time and that would still change with time

Reply:
Ah I see the mock function below. Just use `datetime.now` and mock it within this module to return a fixed datetime instance

Reply:
You would have to mock datetime.datetime itself and then manually replace datetime.now.

in `topic_editor.py`
```python
import datetime
```

and in your test, something like
```
with self.swap(topic_editor.datetime.datetime) as dt:
    print(dt.now()) # calls the real thing
    dt.now.return_value = test_value
    print(dt.now()) # calls the mocked value
```

If this doesn't work with the swap functions, you can use python's `unitteest.mock.patch` to do this. And even if that doesn't work, you could create a new static function wihin the class that just returns the current datetime and mock that function in your test. I think it will be easier to understand than manually adusting timezone after getting a utc timestamp.

```python
@staticmethod
def get_current_datetime():
    return datetime.datetime.now()
```

Since this has already been discussed with @seanlip before, pinging him for his thoughts.

Reply:
Ah, I didn't realize that if you used get_current_time_in_millisecs() it would require additional manual adjustments -- that is indeed a bit messy.

That said, normally I believe we don't handle timezone offsets server-side, especially since the user may be in a different region from the server. Why not send the UTC-timezone data to the client browser, and let that handle any conversion to local time that is needed? In general don't introduce new patterns here if you don't need to.

Also @Kej-r03 I see this pattern in some of your other PRs, it would be good to correct it as well there. Look at creator_dashboard.py, you can pass the time in msecs directly to the frontend and let the frontend handle it.

Reply:
@seanlip +1 to keeping everything utc. In this case, @Kej-r03 mentioned that `Date.now()` on frontend  is used to generate planned_publication_date_msecs which gives us the utc timestamp. That is converted to a datetime object in the server's timezone at the backend so both would have to be changed to utc timestamps.

Reply:
Yup, typically we deal directly with the msec values on the backend as much as possible to avoid all the timezone computations/ambiguity with datetimes.

Reply:
@Kej-r03 But can we avoid any timezone adjustments in the backend codebase at all? That would be better for maintainability.

Reply:
@Kej-r03 I'm not sure if you saw the above, but could you please reply to this?

In general I would advise dealing with msecs (rather than datetimes) throughout the code as much as possible.

Reply:
@seanlip  PTAL now, I have changed it to msecs.